File: languages/js/src/entity/identity_v2/string_attribute.ts
import * as identityEntitiesProto from "../../bridge/proto/identity_entities_v2";
import { Attribute } from "./attribute";

export class StringAttribute extends Attribute<string> {
  public toProto(): identityEntitiesProto.StringAttributeV2 {
    return identityEntitiesProto.StringAttributeV2.fromPartial({
      id: this.id,
      value: this.value
    });
  }

  static fromProto(
    r: identityEntitiesProto.StringAttributeV2
  ): StringAttribute {
    return new StringAttribute(r.id, r.value);
  }
}

File: languages/python/bloock/entity/identity_v2/string_attribute.py
from __future__ import annotations

import bloock._bridge.proto.identity_entities_v2_pb2 as proto
from bloock.entity.identity_v2.attribute import Attribute


class StringAttribute(Attribute):
    @staticmethod
    def from_proto(a: proto.StringAttributeV2) -> StringAttribute:
        return StringAttribute(
            id=a.id,
            value=a.value,
        )

    def to_proto(self) -> proto.StringAttributeV2:
        return proto.StringAttributeV2(
            id=self.id,
            value=self.value,
        )
